### PR TITLE
[PyOV] Optimize dtype to Type conversion

### PR DESCRIPTION
**Pull request by jiwaszki**
### Details:
 - Remove costly conversion to string by replacing it with unique ids of each dtype.
 - More on `dtype.num`: https://numpy.org/devdocs/reference/generated/numpy.dtype.num.html#numpy.dtype.num

### Tickets:
 - *...*

### CODE REVIEW

File: src/bindings/python/src/pyopenvino/core/common.cpp [hunk 1]
```diff
@@ -37,24 +37,34 @@ py::dtype get_dtype(const ov::element::Type& ov_type) {
...
+        {12, ov::element::f64},     // float64
+        {1, ov::element::i8},       // int8
+        {3, ov::element::i16},      // int16
+#ifdef _WIN32
+        {7, ov::element::i32},      // int32
+        {9, ov::element::i64},      // int64
+#else
+        {5, ov::element::i32},      // int32
+        {7, ov::element::i64},      // int64
+#endif
```
Comment (jiwaszki) [inline src/bindings/python/src/pyopenvino/core/common.cpp:53]: Predicting questions: https://github.com/numpy/numpy/issues/8433

Numpy 2.0 should resolve this, until that point this is required.

**Review by p-wysocki — APPROVED** [review]

File: src/bindings/python/src/pyopenvino/core/common.cpp (hunk 1 above)
Comment (ilya-lavrenov) [reply src/bindings/python/src/pyopenvino/core/common.cpp]: such comments are worth to be added to code itself :)

**Review by akuporos — APPROVED** [review]
Wow, good find!

File: tools/mo/unit_tests/mock_mo_frontend/mock_mo_python_api/CMakeLists.txt [hunk 2]
```diff
@@ -26,9 +26,9 @@ set(PYBIND_FE_SRC ${CMAKE_CURRENT_SOURCE_DIR}/mock_mo_python_api.cpp)
 source_group("src" FILES ${PYBIND_FE_SRC})
 
 if(Python3_VERSION_MINOR GREATER_EQUAL 11)
-    set(pybind11_min_version 2.9.2)
+    set(pybind11_min_version 2.10.0)
 else()
-    set(pybind11_min_version 2.8.0)
+    set(pybind11_min_version 2.10.0)
 endif()
```
Comment (mlukasze) [inline tools/mo/unit_tests/mock_mo_frontend/mock_mo_python_api/CMakeLists.txt:32]: is the condition still necessary?

File: src/bindings/python/CMakeLists.txt [hunk 3]
```diff
@@ -199,9 +199,9 @@ endif()
 #
 
 if(Python3_VERSION_MINOR GREATER_EQUAL 11)
-    set(pybind11_min_version 2.9.2)
+    set(pybind11_min_version 2.10.0)
 else()
-    set(pybind11_min_version 2.8.0)
+    set(pybind11_min_version 2.10.0)
```
[the remark this answers is not in the document]
Comment (jiwaszki) [reply src/bindings/python/CMakeLists.txt]: Is it OK now?

File: tools/mo/unit_tests/mock_mo_frontend/mock_mo_python_api/CMakeLists.txt (hunk 2 above)
Comment (jiwaszki) [reply tools/mo/unit_tests/mock_mo_frontend/mock_mo_python_api/CMakeLists.txt]: Is it OK now?

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
@mlukasze I suppose such useful change is worth to be ported to LTS

File: src/bindings/python/src/pyopenvino/core/common.cpp [hunk 4]
```diff
@@ -37,24 +37,34 @@ py::dtype get_dtype(const ov::element::Type& ov_type) {
...
+        {11, ov::element::f32},  // float32
+        {12, ov::element::f64},  // float64
+        {1, ov::element::i8},    // int8
+        {3, ov::element::i16},   // int16
+#ifdef _WIN32
+        {7, ov::element::i32},  // int32
+        {9, ov::element::i64},  // int64
+#else
+        {5, ov::element::i32},  // int32
+        {7, ov::element::i64},  // int64
```
Comment (rkazants) [inline src/bindings/python/src/pyopenvino/core/common.cpp:52]: why are we sure that these numbers are valid for different platforms and NumPy versions?

**Pull request merged**